waddrmgr/db.go

```diff
@@ -770,9 +841,11 @@ func fetchAddress(tx walletdb.Tx, addressID []byte) (interface{}, error) {
 	case adtChain:
 		return deserializeChainedAddress(addressID, row)
 	case adtImport:
-		return deserializeImportedAddress(addressID, row)
+		return deserializeImportedPubKeyAddress(addressID, row)
 	case adtScript:
 		return deserializeScriptAddress(addressID, row)
+	case adtWatching:
+		return deserializeImportedHash160AddressRow(addressID, row)
```

Comment:
Given the above comments, this line will end up going away with the removal of the `adtWatching` type.

Of note is there is already code to deal with watching-only address conversion in `deletePrivateKeys` which is used when converting the entire manager to watching-only, so, outside of the addition of the watching-only flag to each address, there shouldn't be any need to modify the raw data serialization formats at all.